contracts/ERC721A.sol

```diff
@@ -504,6 +510,65 @@ contract ERC721A is IERC721A {
         _afterTokenTransfers(address(0), to, startTokenId, quantity);
     }
 
+    /**
+     * @dev Mints `quantity` tokens and transfers them to `to`.
+     *
+     * This function is intended for efficient minting during contract creation.
+     *
+     * It emits only one {ConsecutiveTransfer} as defined in
+     * [ERC2309](https://eips.ethereum.org/EIPS/eip-2309),
+     * instead of a sequence of {Transfer} event(s).
+     *
+     * Calling this function outside of contract creation WILL break the ERC721 standard.
+     * For full ERC721 compliance, substituting ERC721 {Transfer} event(s) with the ERC2309
+     * {ConsecutiveTransfer} event is only permissible during contract creation.
+     *
+     * Note: some marketplaces may have difficulties fully registering a very large `quantity`.
+     * We have verified full support on OpenSea's and LooksRare's Rinkeby testnet marketplaces
+     * for `quantity = 5000` on 5th June 2022. Support may vary, and may change over time.
+     *
+     * Requirements:
+     *
+     * - `to` cannot be the zero address.
+     * - `quantity` must be greater than 0.
+     *
+     * Emits a {ConsecutiveTransfer} event.
+     */
+    function _mintERC2309(address to, uint256 quantity) internal {
+        uint256 startTokenId = _currentIndex;
+        if (_addressToUint256(to) == 0) revert MintToZeroAddress();
+        if (quantity == 0) revert MintZeroQuantity();
+
+        _beforeTokenTransfers(address(0), to, startTokenId, quantity);
+
+        // Overflows are incredibly unrealistic.
```

Comment:
⚠️  Security issue.

Overflows are incredibly easy and they result in data corruption across the bitfield, and inconsistent results in `_afterTokenTransfers` versus the quantity actually minted.

For example, this function allows to effectively mint negative one token by passing a value of 2^255.

Reply:
Added a max `quantity` limit of 5000 per call to `_mintERC2309`.